### PR TITLE
Improve node names and handling of duplicates

### PR DESCRIPTION
**Pull request by dimaryaz**
- If the filename starts with a number, use the "n" prefix
- When using file extensions to resolve conflicts, add them to all of the duplicates
- Don't use more than one `_` in a row

### CODE REVIEW

File: quilt/tools/build.py [hunk 1]
```diff
@@ -23,15 +25,14 @@ def _is_internal_node(node):
     return all(isinstance(x, dict) for x in node.values())
 
 def _pythonize_name(name):
-    safename = re.sub('[^A-Za-z0-9_]+', '_', name)
-    starts_w_number = re.match('^[0-9].*', safename)
-    if starts_w_number:
-        safename = ("abc%s" % safename)
-
+    safename = re.sub('[^A-Za-z0-9]+', '_', name)
     safename = safename.strip('_')
```
Comment (akarve) [inline quilt/tools/build.py]: very minor. why not chain the above two calls? maybe that's not our codebase style just asking.

File: quilt/tools/build.py [hunk 2]
```diff
@@ -177,49 +178,66 @@ def splitext_no_dot(filename):
...
+        for safename, name in iteritems(safename_to_name):
+            path = os.path.join(dir_path, name)
+
+            if os.path.isdir(path):
+                data = _generate_contents(path)
+            else:
+                rel_path = os.path.relpath(path, startpath)
+                data = dict(file=rel_path)
+
+            contents[safename] = data
```
Comment (akarve) [inline quilt/tools/build.py]: should we backlog unit tests for this or is existing coverage sufficient?

File: quilt/tools/build.py (hunk 1 above)
Comment (dimaryaz) [inline quilt/tools/build.py]: Sure

File: quilt/tools/build.py (hunk 2 above)
Comment (dimaryaz) [inline quilt/tools/build.py]: Definitely need unit tests; even if there's existing coverage, it doesn't cover the bugs I'm fixing here.

File: quilt/test/test_build.py [hunk 3]
```diff
@@ -96,3 +97,49 @@ def test_failover(self):
         path = os.path.join(mydir, './build_failover.yml')
         build.build_package('test_failover', PACKAGE, path)
         from quilt.data.test_failover.groot import bad
+
+    def test_duplicates(self):
```
Comment (kevinemoore) [inline quilt/test/test_build.py]: This test is failing for me in Python 3.5.3:

E       AssertionError: assert {'a_b_c': {'f...xt.csv'}, ...} == {'a': {}, 'a_b...'a.txt'}, ...}
E         Common items:
E         {'a_b_c': {'file': 'a%%b___c'},
E          'a_csv': {'file': 'a.csv'},
E          'a_txt': {'file': 'a.txt'},
E          'a_txt_csv': {'file': 'a_txt.csv'},
E          'n1': {'file': '1'},
E          'n123': {'file': '_123'},
E          'n1_txt': {'file': '1.txt'}}
E         Right contains more items:
E         {'a': {}, 'dir_ext': {}}
E         Full diff:
E         + {'a': {},
E         - {'a_b_c': {'file': 'a%%b___c'},
E         ? ^
E         +  'a_b_c': {'file': 'a%%b___c'},
E         ? ^
E         'a_csv': {'file': 'a.csv'},
E         'a_txt': {'file': 'a.txt'},
E         'a_txt_csv': {'file': 'a_txt.csv'},
E         +  'dir_ext': {},
E         'n1': {'file': '1'},
E         'n123': {'file': '_123'},
E         'n1_txt': {'file': '1.txt'}}

Comment (dimaryaz) [inline quilt/test/test_build.py]: Oh... I think it's because git doesn't allow empty directories. Let me fix that.

**Pull request merged**